### PR TITLE
Deterministic symmetric key encryption #5

### PR DESCRIPTION
**Pull request by lamp**
#5

### CODE REVIEW

File: src/tinklj/config.clj [hunk 1]
```diff
@@ -8,28 +8,19 @@
...
+  "Initialise tink, optionally you can provide a type
+  to only initialise a type, for example :aead
+  Returns: nil
+  Example:
+    (register :mac)"
+  [& config-type]
+  (case (first config-type)
+   :aead (AeadConfig/register)
+   :daead (DeterministicAeadConfig/register)
+   :mac (MacConfig/register)
```
Comment (perkss) [inline src/tinklj/config.clj]: 🥇

**Comment by perkss** [comment]
@lamp be good to note down the java version that you had problems with

File: test/tinklj/acceptance/symmetric_deterministic_key_encryption_test.clj [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+(register :daead)
+
+(deftest symmetric-deterministic-key-encryption
+
+  (testing "Deterministic Encryption and Decryption"
+      (let [aad "salt"
+            handle (keyset/generate-new :aes256-siv)
+            primitive (deterministic handle)
+            decryptable-data (sut/encrypt primitive
```
Comment (perkss) [inline test/tinklj/acceptance/symmetric_deterministic_key_encryption_test.clj]: wouldnt this be better named encrypted data?

File: src/tinklj/daead.clj [hunk 3]
```diff
@@ -0,0 +1,15 @@
+(ns tinklj.daead
+  (:require [tinklj.keys.keyset-handle :as keyset-handles])
+  (:import (java.security Security)))
+
+(Security/setProperty "crypto.policy" "unlimited")
```
Comment (perkss) [inline src/tinklj/daead.clj]: what does this do?

Comment (lamp) [reply src/tinklj/daead.clj]: Oh, this allows unlimited encryption to be used: https://stackoverflow.com/questions/24907530/java-security-invalidkeyexception-illegal-key-size-or-default-parameters-in-and

(Basically why everything was broken for so long on my branch)

Comment (lamp) [reply src/tinklj/daead.clj]: Turns out it wasn't needed, removed.

**Pull request merged**